Document the workflow contract for ecosystem packages (#145) (#152)
Gilead-BioStats/gsm.core: README.md

-   **Function Documentation** - Comprehensive roxygen2 documentation maintained for all functions.
-   **Data Specifications** - Machine-readable specifications for all KRIs, automatically integrated into function documentation.

> -   **Contributor Guidelines** - Step-by-step processes for code development, releases, and workflow YAML conventions are provided in [CONTRIBUTING.md](.github/CONTRIBUTING.md) and required to be followed for contributions.

Reviewer: ```suggestion
-   **Contributor Guidelines** - Step-by-step processes for code development, releases, and workflow YAML conventions are provided in [CONTRIBUTING.md](.github/CONTRIBUTING.md) and must be followed for contributions.
```

-   **Function Documentation** - Comprehensive roxygen2 documentation maintained for all functions.
-   **Data Specifications** - Machine-readable specifications for all KRIs, automatically integrated into function documentation.
-   **Contributor Guidelines** - Step-by-step processes for code development and releases provided in `CONTRIBUTING.md` and required to be followed for contributions.

> -   **Workflow Contract** - Conventions ecosystem packages must follow when shipping workflow YAMLs (location, numbered phases, naming, and collision rules) are documented in the [Workflow Contract for Ecosystem Packages](.github/CONTRIBUTING.md#workflow-contract-for-ecosystem-packages) section of `CONTRIBUTING.md`.

Reviewer: It doesn't feel like this needs its own bullet, since it's part of `CONTRIBUTING.md`.
